frame/offences/src/migration.rs

```diff
@@ -0,0 +1,99 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::{Config, OffenceDetails, Perbill, SessionIndex};
+use frame_support::{traits::Get, weights::Weight, generate_storage_alias};
+use sp_staking::offence::OnOffenceHandler;
+use sp_std::vec::Vec;
+
+/// Type of data stored as a deferred offence
+type DeferredOffenceOf<T> = (
+	Vec<
+		OffenceDetails<
+			<T as frame_system::Config>::AccountId,
+			<T as Config>::IdentificationTuple,
+		>,
+	>,
```

Comment:
It looks like there is another migration being introduced here cc @kianenigma

Reply:
Does this need to be executed? can we check that it is will be executed?

Reply:
this migration is pretty harmless. Even if we leave it in the code and re-execute it again later it will be fine. It will just remove a storage item that no longer exists and has no reference in the code

Reply:
fwiw i went through substrate commits on monday to check for diffs to migrations code since 0.8.30, this was the only one i saw. it was removed with 57402708ebbf17de587fecba3db525d7c66707c3.